### ISSUE TITLE
Enhancement request for Tab component to support disabled tabs with tooltips

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-design#1066]
A use case for having a disabled tab has come up in recent discussions. This use case is similar to the one where an action displays but is disabled with a tooltip explaining why the user can't access the action. In the case of tabs, the tab includes contents which are not available to the user due to lack of permission to access the contents.

We did discuss the approach to enable the tab and display messaging in an empty state. The general consensus was that if we can tell them sooner and avoid the click and page loading an empty state, that it would be a nicer experience.

Something to keep in mind if this design goes to core and react is that the implementation we have for applying tooltips to disabled buttons would need to be supported here as well (i.e. `aria-disabled` instead of `disabled` so that the tab can receive focus and display the tooltip message).

The timing for this is September 2021, and would be used for apps (most likely Apicurio) under Application Services in clouddot. 

cc'ing @lboehling and @dlabaj

**Comment by mcarrano** [comment]
@jgiardino Seems like a reasonable thing to support. Delivering this by September will be challenging given that we don't currently have disabled states for tabs. I think best case I can try to slot this into our September 15 release. Will that work?

@lboehling would you be able to pick up the design work in the upcoming PatternFly sprint (starting next week)?

**Comment by jgiardino** [comment]
Thanks for the update, @mcarrano!

@dlabaj What do you think about the timing?

The September timeframe is for early eval, whereas GA is targeted for Nov.

**Comment by lboehling** [comment]
@mcarrano I can take on this issue for this sprint!

**Comment by lboehling** [comment]
Here is my design suggestion for disabled state styling across PF tabs--

![image](https://user-images.githubusercontent.com/46571616/127548848-41e2b45a-7847-47e3-8ab4-f969fec7d815.png)


@mcarrano @jgiardino @dlabaj

**Comment by mcarrano** [comment]
@lboehling I think this works. Are these the same gray values for disabled that we are using for buttons?

@mceledonia @mcoker can you also review?

**Comment by mcoker** [comment]
@mcarrano lgtm

**Comment by mcarrano** [comment]
I've opened a new core issue here: https://github.com/patternfly/patternfly/issues/4257

**Comment by mceledonia** [comment]
These look good to me!

**Comment by mcarrano** [comment]
I'm going to call this complete then. Thanks for your work on this @lboehling !

**Comment by jhiadlov** [comment]
@mcarrano Hello, I have been working on this issue in the Sketch library, and I would like to know, should I also add a disabled state for horizontal tabs with icons? :)

**Comment by mcarrano** [comment]
Yes. No reason that disabled tabs can't work with or without icons.
